src/main/java/org/takes/facets/auth/Token.java

```diff
@@ -0,0 +1,187 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.TimeZone;
+import javax.json.Json;
+import javax.json.JsonObject;
+import org.takes.misc.Base64;
+
+/**
+ * JSON Token.
+ *
+ * <p>
+ * All implementations of this interface must be immutable and thread-safe.
+ *
+ * @author Sven Windisch (sven.windisch@gmail.com)
+ * @version $Id$
+ * @since 1.4
+ */
+public interface Token {
+
+    /**
+     * JSON output.
+     *
+     * @return The Token in JSON notation.
+     */
+    JsonObject json();
+
+    /**
+     * Base64 encoded JSON output.
+     *
+     * @return The Token in JSON notation, Base64-encoded.
+     */
+    byte[] encoded();
+
+    /**
+     * JSON Object Signing and Encryption Header.
+     */
+    final class Jose implements Token {
+        /**
+         * The header short for algorithm.
+         */
+        public static final String ALGORITHM = "alg";
+
+        /**
+         * The header short for token type.
+         */
+        public static final String TYP = "typ";
+
+        /**
+         * JOSE object.
+         */
+        private final JsonObject joseo;
+
+        /**
+         * JSON Object Signing and Encryption Header.
+         * @param bitlength Of encryption bits.
+         */
+        public Jose(final int bitlength) {
+            this.joseo = Json.createObjectBuilder()
+                .add(Jose.ALGORITHM, String.format("HS%s", bitlength))
+                .add(Jose.TYP, "JWT")
+                .build();
+        }
+
+        @Override
+        public JsonObject json() {
+            return this.joseo;
+        }
+
+        @Override
+        public byte[] encoded() {
+            byte[] bytes;
+            try {
+                bytes = new Base64().encode(this.joseo.toString());
+            } catch (final IOException ignore) {
+                bytes = new byte[] {};
+            }
+            return bytes;
+        }
+    };
+
+    /**
+     * JSON Web Token.
+     */
+    @SuppressWarnings
```

Comment:
@semantosoph you can remove that and just initialize all attributes in the constructor, it's a good practice.